src/sagemaker/jumpstart/curated_hub/utils.py

```diff
@@ -164,3 +182,121 @@ def create_hub_bucket_if_it_does_not_exist(
+        version=version,
+        region=region,
+        scope=JumpStartScriptScope.INFERENCE,
+        tolerate_vulnerable_model = True,
+        tolerate_deprecated_model = True,
+        sagemaker_session=session,
+    )
+
+    if (jumpstart_model_specs.deprecated):
+        tags_to_add.append(CuratedHubTagName.DEPRECATED_VERSIONS)
+    if (jumpstart_model_specs.inference_vulnerable):
+        tags_to_add.append(CuratedHubTagName.INFERENCE_VULNERABLE_VERSIONS)
+    if (jumpstart_model_specs.training_vulnerable):
+        tags_to_add.append(CuratedHubTagName.TRAINING_VULNERABLE_VERSIONS)
+
+    return tags_to_add
+
+    
+
+def get_jumpstart_model_and_version(hub_content_summary: HubContentSummary) -> Optional[JumpStartModelInfo]:
+    jumpstart_model_id_tag = next(
+        (
+            tag
+            for tag in hub_content_summary.hub_content_search_keywords
+            if tag.startswith(JUMPSTART_HUB_MODEL_ID_TAG_PREFIX)
+        ),
+        None,
+    )
+    jumpstart_model_version_tag = next(
+        (
+            tag
+            for tag in hub_content_summary.hub_content_search_keywords
+            if tag.startswith(JUMPSTART_HUB_MODEL_VERSION_TAG_PREFIX)
+        ),
+        None,
+    )
+
+    if jumpstart_model_id_tag is None or jumpstart_model_version_tag is None:
+        return None
+    jumpstart_model_id = jumpstart_model_id_tag[len(JUMPSTART_HUB_MODEL_ID_TAG_PREFIX):] # Need to remove the tag_prefix and ":"
+    jumpstart_model_version = jumpstart_model_version_tag[len(JUMPSTART_HUB_MODEL_ID_TAG_PREFIX):]
+    return JumpStartModelInfo(model_id=jumpstart_model_id, version=jumpstart_model_version)
+
+def summary_from_list_api_response(hub_content_summary: Dict[str, Any]) -> HubContentSummary:
+    return HubContentSummary(
+        hub_content_arn=hub_content_summary.get("HubContentArn"),
+        hub_content_name=hub_content_summary.get("HubContentName"),
+        hub_content_version=hub_content_summary.get("HubContentVersion"),
+        hub_content_type=hub_content_summary.get("HubContentType"),
+        document_schema_version=hub_content_summary.get("DocumentSchemaVersion"),
+        hub_content_status=hub_content_summary.get("HubContentStatus"),
+        hub_content_display_name=hub_content_summary.get("HubContentDisplayName"),
+        hub_content_description=hub_content_summary.get("HubContentDescription"),
+        hub_content_search_keywords=hub_content_summary.get("HubContentSearchKeywords"),
+        creation_time=hub_content_summary.get("CreationTime")
+    )
+
+def summary_list_from_list_api_response(list_hub_contents_response: Dict[str, Any]) -> List[HubContentSummary]:
+    return list(map(summary_from_list_api_response, list_hub_contents_response["HubContentSummaries"]))
+
+def tag_to_add_tags_api_call(tag: CuratedHubTag) -> Dict[str, str]:
+    return {
+          'Key': tag.key,
+          'Value': tag.value
+    }
+
+def generate_unique_hub_content_model_name(model_id: str) -> str:
```

Comment:
I don't see this being used, can we remove? Not sure what this would be used for..